cassiopeia/data/utilities.py

```diff
@@ -165,4 +174,131 @@ def _compute_dissimilarity_map(cm, C, missing_state_indicator, nb_weights):
 
         return dm
 
-    return _compute_dissimilarity_map(cm, C, missing_state_indicator, nb_weights)
+    return _compute_dissimilarity_map(
+        cm, C, missing_state_indicator, nb_weights
+    )
+
+
+def sample_bootstrap_character_matrices(
+    character_matrix: pd.DataFrame,
+    prior_probabilities: Optional[Dict[int, Dict[int, float]]] = None,
+    B: int = 10,
+    random_state: Optional[np.random.RandomState] = None,
+) -> List[Tuple[pd.DataFrame, Dict[int, Dict[int, float]]]]:
```

Comment:
`new_priors` can be `None`, so the return type should be `List[Tuple[pd.DataFrame, Optional[Dict[int, Dict[int, float]]]]]`.